packages/testlab/src/http-error-logger.ts

```diff
@@ -0,0 +1,24 @@
+import {Request} from 'express';
+import {IncomingMessage} from 'http';
+
+export function createUnexpectedHttpErrorLogger(
+  expectedStatusCode: number = 0,
```

Comment:
Should we use `200` as the default value?

Reply:
Can we accept `...expectedStatusCodes: number[]` to allow multiple status codes without breaking existing code?

Reply:
The purpose of this helper is to disable error logs printed by our default error handler for the status code expected by an integration/end-to-end test making HTTP calls to the tested application.

Typically, such test is verifying the response status code against a single value only.

```ts
await client.get('/foo/bar').expect(404);
```

> Should we use `200` as the default value?

That does not make much sense to me - no error is logged for 2xx and 3xx status codes. This helper is useful only for tests expecting 4xx or 5xx response.

> Can we accept `...expectedStatusCodes: number[]` to allow multiple status codes without breaking existing code?

I think so. However, I am not convinced this a good practice to support. Let's defer this enhancement until we have a real-world use case needing it.

Reply:
> That does not make much sense to me - no error is logged for 2xx and 3xx status codes. This helper is useful only for tests expecting 4xx or 5xx response.

Then why don't we have the following signature? It leaves the default to `undefined` instead of `0` to avoid similar confusions as I had.
```ts
export function createUnexpectedHttpErrorLogger(
  expectedStatusCode? number,
) {}
```

Reply:
I don't remember why I picked `0` as the default value. Your proposal to use `undefined` instead of `0` is fine with me 👍

Reply:
Updated to use `undefined`. 👍